test/test_torch.py

```diff
@@ -17499,10 +17498,14 @@ def test_fmod(self, device, dtype):
 
         zero = torch.zeros_like(m1)
         if dtype in torch.testing.get_all_int_dtypes():
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                m1.fmod(0)
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                m1.fmod(zero)
+            if device == 'cpu':
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    m1.fmod(0)
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    m1.fmod(zero)
+            else:
```

Comment:
(Putting this comment here, but it's unrelated to the code) could we add some test cases for what happens if the dtypes aren't the same? I'd want to make sure that the error messages and behavior are the same before and after this PR. 

For example, with 
```
x = torch.tensor([1, 2, 3, 4])
y = torch.randn(4)
torch.fmod(x, y)
```
we get "result type Float can't be cast to the desired output type Long"

Reply:
Yes, it makes sense!